src/Renci.SshNet/BaseClient.cs

```diff
@@ -37,6 +35,14 @@ public abstract class BaseClient : IBaseClient
         /// </value>
         internal ISession? Session { get; private set; }
 
+        internal ILogger? Logger
+        {
+            get
+            {
+                return Session?.SessionLoggerFactory?.CreateLogger(GetType());
+            }
+        }
+
```

Comment:
This could be made non-nullable, so the `Logger?.` could be avoided as well.

```suggestion
        internal ILogger Logger
        {
            get
            {
                return Session?.SessionLoggerFactory?.CreateLogger(GetType()) ?? NullLogger.Instance;
            }
        }

```

Reply:
Thanks. Solved by implementing the change after @Rob-Hague comment.